pkg/backend/snapshot.go

```diff
@@ -507,7 +496,7 @@ func (sm *SnapshotManager) snap() *deploy.Snapshot {
 	manifest := deploy.Manifest{
 		Time:    time.Now(),
 		Version: version.Version,
-		Plugins: sm.plugins,
+		// Plugins: sm.plugins, - Explicitly dropped, since we don't use the plugin list in the manifest anymore.
```

Comment:
Is there a compelling reason for retaining the `Plugins` field of `deploy.Manifest` at all?

Reply:
I don't think so - as long as this is an engine-internal type (which I'm pretty sure it is?) it should be fine.

Reply:
Turns out there is a compelling reason:

When the engine encounters a stack whose snapshot doesn't have any default provider references, it injects them as necessary:

https://github.com/pulumi/pulumi/blob/0550f71a35df8bce291d299f5b4c653968f1fe8c/pkg/resource/deploy/plan.go#L111-L115

In order to do this, it uses the manifest to figure out what version of the plugin to inject into the first-class provider resource that it synthesizes. I don't think we can safely remove this code without breaking users using this CLI to update a very old stack.

Reply:
Makes sense!